ktor-client/ktor-client-core/common/src/io/ktor/client/request/utils.kt

```diff
@@ -77,3 +78,15 @@ public fun HttpRequestBuilder.parameter(key: String, value: Any?): Unit =
  */
 public fun HttpRequestBuilder.accept(contentType: ContentType): Unit =
     headers.append(HttpHeaders.Accept, contentType.toString())
+
+/**
+ * Sets the [HttpHeaders.Authorization] to Basic Authorization with the provided [username] and [password]
+ */
+public fun HttpRequestBuilder.basicAuth(username: String, password: String): Unit =
```

Comment:
Would it be better to put it in the same place with `Bearer/BasicAuthProvider`?

Reply:
This would require adding the `auth` artifact too. 
My idea was to provide a shortcut function for simple one-time usage. For more configs/options (like default auth on every request) you should use the `auth` artifact.